## Week 2

### Exercise:
Use and experience methods from previous lessons with given data. Try to search for other pandas methods to analyze the data such as `groupby()` and `read_csv()`.

*Verilen verilerle önceki derslerdeki yöntemleri kullanın ve deneyimleyin. Verileri analiz etmek için "groupby()" ve "read_csv()" gibi diğer pandas yöntemlerini aramayı deneyin.*

In [1]:
# You can find covid related data under project's data folder 
base_url = "../data/covid/"
infected_dataset_url = base_url + "time_series_covid19_confirmed_global.csv"
recovered_dataset_url = base_url + "time_series_covid19_recovered_global.csv"
deaths_dataset_url = base_url + "time_series_covid19_deaths_global.csv"

#### Solution:

In [2]:
import pandas as pd

**IMPORTING DATASETS**

In [3]:
infected_dataset = pd.read_csv(infected_dataset_url)

In [4]:
infected_dataset

Province/State         Country/Region        Lat       Long  1/22/20  \
0              NaN            Afghanistan  33.000000  65.000000        0   
1              NaN                Albania  41.153300  20.168300        0   
2              NaN                Algeria  28.033900   1.659600        0   
3              NaN                Andorra  42.506300   1.521800        0   
4              NaN                 Angola -11.202700  17.873900        0   
..             ...                    ...        ...        ...      ...   
261            NaN  Sao Tome and Principe   0.186360   6.613081        0   
262            NaN                  Yemen  15.552727  48.516388        0   
263            NaN                Comoros -11.645500  43.333300        0   
264            NaN             Tajikistan  38.861034  71.276093        0   
265            NaN                Lesotho -29.609988  28.233608        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  6/6/20  6/7/20  6/8/20  \
0          0        0        0        0        0  ...   19551   20342   20917   
1          0        0        0        0        0  ...    1232    1246    1263   
2          0        0        0        0        0  ...   10050   10154   10265   
3          0        0        0        0        0  ...     852     852     852   
4          0        0        0        0        0  ...      88      91      92   
..       ...      ...      ...      ...      ...  ...     ...     ...     ...   
261        0        0        0        0        0  ...     499     513     513   
262        0        0        0        0        0  ...     482     484     496   
263        0        0        0        0        0  ...     141     141     141   
264        0        0        0        0        0  ...    4453    4529    4609   
265        0        0        0        0        0  ...       4       4       4   

     6/9/20  6/10/20  6/11/20  6/12/20  6/13/20  6/14/20  6/15/20  
0     21459    22142    22890    23546    24102    24766    25527  
1      1299     1341     1385     1416     1464     1521     1590  
2     10382    10484    10589    10698    10810    10919    11031  
3       852      852      852      853      853      853      853  
4        96      113      118      130      138      140      142  
..      ...      ...      ...      ...      ...      ...      ...  
261     514      611      632      639      659      661      662  
262     524      560      591      632      705      728      844  
263     141      162      162      163      176      176      176  
264    4690     4763     4834     4902     4971     5035     5097  
265       4        4        4        4        4        4        4  

[266 rows x 150 columns]

In [5]:
recovered_dataset = pd.read_csv(recovered_dataset_url)

In [6]:
recovered_dataset

Province/State         Country/Region        Lat       Long  1/22/20  \
0              NaN            Afghanistan  33.000000  65.000000        0   
1              NaN                Albania  41.153300  20.168300        0   
2              NaN                Algeria  28.033900   1.659600        0   
3              NaN                Andorra  42.506300   1.521800        0   
4              NaN                 Angola -11.202700  17.873900        0   
..             ...                    ...        ...        ...      ...   
248            NaN  Sao Tome and Principe   0.186360   6.613081        0   
249            NaN                  Yemen  15.552727  48.516388        0   
250            NaN                Comoros -11.645500  43.333300        0   
251            NaN             Tajikistan  38.861034  71.276093        0   
252            NaN                Lesotho -29.609988  28.233608        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  6/6/20  6/7/20  6/8/20  \
0          0        0        0        0        0  ...    1830    1875    2171   
1          0        0        0        0        0  ...     925     938     945   
2          0        0        0        0        0  ...    6631    6717    6799   
3          0        0        0        0        0  ...     741     744     751   
4          0        0        0        0        0  ...      24      24      38   
..       ...      ...      ...      ...      ...  ...     ...     ...     ...   
248        0        0        0        0        0  ...      68      68      68   
249        0        0        0        0        0  ...      23      23      23   
250        0        0        0        0        0  ...      67      67      67   
251        0        0        0        0        0  ...    2583    2673    2763   
252        0        0        0        0        0  ...       2       2       2   

     6/9/20  6/10/20  6/11/20  6/12/20  6/13/20  6/14/20  6/15/20  
0      2651     3013     3326     3928     4201     4725     5164  
1       960      980     1001     1034     1039     1044     1055  
2      6951     7074     7255     7322     7420     7606     7735  
3       757      759      780      781      781      781      789  
4        38       40       41       42       61       61       64  
..      ...      ...      ...      ...      ...      ...      ...  
248      76      135      146      156      176      177      177  
249      23       23       23       28       39       53       79  
250      67       97       97       97      114      114      114  
251    2815     2947     3062     3158     3288     3409     3503  
252       2        2        2        2        2        2        2  

[253 rows x 150 columns]

In [7]:
deaths_dataset = pd.read_csv(deaths_dataset_url)

In [8]:
deaths_dataset

Province/State         Country/Region        Lat       Long  1/22/20  \
0              NaN            Afghanistan  33.000000  65.000000        0   
1              NaN                Albania  41.153300  20.168300        0   
2              NaN                Algeria  28.033900   1.659600        0   
3              NaN                Andorra  42.506300   1.521800        0   
4              NaN                 Angola -11.202700  17.873900        0   
..             ...                    ...        ...        ...      ...   
261            NaN  Sao Tome and Principe   0.186360   6.613081        0   
262            NaN                  Yemen  15.552727  48.516388        0   
263            NaN                Comoros -11.645500  43.333300        0   
264            NaN             Tajikistan  38.861034  71.276093        0   
265            NaN                Lesotho -29.609988  28.233608        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  6/6/20  6/7/20  6/8/20  \
0          0        0        0        0        0  ...     327     357     369   
1          0        0        0        0        0  ...      34      34      34   
2          0        0        0        0        0  ...     698     707     715   
3          0        0        0        0        0  ...      51      51      51   
4          0        0        0        0        0  ...       4       4       4   
..       ...      ...      ...      ...      ...  ...     ...     ...     ...   
261        0        0        0        0        0  ...      12      12      12   
262        0        0        0        0        0  ...     111     112     112   
263        0        0        0        0        0  ...       2       2       2   
264        0        0        0        0        0  ...      48      48      48   
265        0        0        0        0        0  ...       0       0       0   

     6/9/20  6/10/20  6/11/20  6/12/20  6/13/20  6/14/20  6/15/20  
0       384      405      426      446      451      471      478  
1        34       34       35       36       36       36       36  
2       724      732      741      751      760      767      777  
3        51       51       51       51       51       51       51  
4         4        4        5        5        6        6        6  
..      ...      ...      ...      ...      ...      ...      ...  
261      12       12       12       12       12       12       12  
262     127      129      136      139      160      164      208  
263       2        2        2        2        2        2        2  
264      48       48       49       49       50       50       50  
265       0        0        0        0        0        0        0  

[266 rows x 150 columns]

**DATASET BRIEF OVERVIEW**

In [9]:
infected_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 266 entries, 0 to 265
Columns: 150 entries, Province/State to 6/15/20
dtypes: float64(2), int64(146), object(2)
memory usage: 311.8+ KB


In [10]:
infected_dataset.describe()

Lat        Long     1/22/20     1/23/20     1/24/20     1/25/20  \
count  266.000000  266.000000  266.000000  266.000000  266.000000  266.000000   
mean    21.148043   22.538640    2.086466    2.458647    3.537594    5.390977   
std     24.909990   70.466256   27.279200   27.377862   34.083035   47.434934   
min    -51.796300 -135.000000    0.000000    0.000000    0.000000    0.000000   
25%      6.907750  -18.093125    0.000000    0.000000    0.000000    0.000000   
50%     23.488100   20.972650    0.000000    0.000000    0.000000    0.000000   
75%     41.143200   77.191525    0.000000    0.000000    0.000000    0.000000   
max     71.706900  178.065000  444.000000  444.000000  549.000000  761.000000   

           1/26/20      1/27/20      1/28/20      1/29/20  ...        6/6/20  \
count   266.000000   266.000000   266.000000   266.000000  ...  2.660000e+02   
mean      7.962406    11.003759    20.969925    23.180451  ...  2.594981e+04   
std      66.289178    89.313757   219.187744   220.524977  ...  1.333309e+05   
min       0.000000     0.000000     0.000000     0.000000  ...  0.000000e+00   
25%       0.000000     0.000000     0.000000     0.000000  ...  1.572500e+02   
50%       0.000000     0.000000     0.000000     0.000000  ...  9.535000e+02   
75%       0.000000     0.000000     0.000000     0.000000  ...  7.094500e+03   
max    1058.000000  1423.000000  3554.000000  3554.000000  ...  1.926538e+06   

             6/7/20        6/8/20        6/9/20       6/10/20       6/11/20  \
count  2.660000e+02  2.660000e+02  2.660000e+02  2.660000e+02  2.660000e+02   
mean   2.637496e+04  2.676449e+04  2.722817e+04  2.773058e+04  2.825084e+04   
std    1.349130e+05  1.363934e+05  1.382464e+05  1.402889e+05  1.424252e+05   
min    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
25%    1.572500e+02  1.572500e+02  1.580000e+02  1.625000e+02  1.650000e+02   
50%    9.710000e+02  9.820000e+02  9.825000e+02  9.825000e+02  9.830000e+02   
75%    7.117250e+03  7.427000e+03  7.706750e+03  8.005750e+03  8.194000e+03   
max    1.944309e+06  1.961781e+06  1.979868e+06  2.000702e+06  2.023590e+06   

            6/12/20       6/13/20       6/14/20       6/15/20  
count  2.660000e+02  2.660000e+02  2.660000e+02  2.660000e+02  
mean   2.873782e+04  2.924391e+04  2.974596e+04  3.020474e+04  
std    1.445871e+05  1.467117e+05  1.484660e+05  1.502466e+05  
min    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  
25%    1.710000e+02  1.722500e+02  1.760000e+02  1.760000e+02  
50%    1.005000e+03  1.005000e+03  1.005000e+03  1.005000e+03  
75%    8.334500e+03  8.452500e+03  8.586500e+03  8.608750e+03  
max    2.048986e+06  2.074526e+06  2.094058e+06  2.114026e+06  

[8 rows x 148 columns]

In [11]:
infected_dataset.shape

(266, 150)

In [12]:
recovered_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 253 entries, 0 to 252
Columns: 150 entries, Province/State to 6/15/20
dtypes: float64(2), int64(146), object(2)
memory usage: 296.6+ KB


In [13]:
recovered_dataset.describe()

Lat        Long     1/22/20     1/23/20     1/24/20     1/25/20  \
count  253.000000  253.000000  253.000000  253.000000  253.000000  253.000000   
mean    19.801381   28.168223    0.110672    0.118577    0.142292    0.154150   
std     24.558602   67.091761    1.760345    1.764332    1.954723    2.020714   
min    -51.796300 -106.346800    0.000000    0.000000    0.000000    0.000000   
25%      6.428100   -7.692100    0.000000    0.000000    0.000000    0.000000   
50%     21.694000   23.881300    0.000000    0.000000    0.000000    0.000000   
75%     39.305400   85.240100    0.000000    0.000000    0.000000    0.000000   
max     71.706900  178.065000   28.000000   28.000000   31.000000   32.000000   

          1/26/20     1/27/20     1/28/20     1/29/20  ...         6/6/20  \
count  253.000000  253.000000  253.000000  253.000000  ...     253.000000   
mean     0.205534    0.241107    0.422925    0.498024  ...   12200.462451   
std      2.649732    2.852382    5.059859    5.566071  ...   45385.270037   
min      0.000000    0.000000    0.000000    0.000000  ...       0.000000   
25%      0.000000    0.000000    0.000000    0.000000  ...      93.000000   
50%      0.000000    0.000000    0.000000    0.000000  ...     586.000000   
75%      0.000000    0.000000    0.000000    0.000000  ...    2971.000000   
max     42.000000   45.000000   80.000000   88.000000  ...  500849.000000   

              6/7/20         6/8/20         6/9/20        6/10/20  \
count     253.000000     253.000000     253.000000     253.000000   
mean    12418.221344   13017.422925   13342.561265   13655.363636   
std     46080.182589   49367.550747   50619.617774   51853.086506   
min         0.000000       0.000000       0.000000       0.000000   
25%        93.000000      98.000000      98.000000      98.000000   
50%       596.000000     603.000000     621.000000     651.000000   
75%      2972.000000    2973.000000    2973.000000    3013.000000   
max    506367.000000  518522.000000  524855.000000  533504.000000   

             6/11/20        6/12/20        6/13/20        6/14/20  \
count     253.000000     253.000000     253.000000     253.000000   
mean    13994.845850   14309.928854   14649.616601   14929.371542   
std     53053.696884   54201.624274   55395.919475   56272.526749   
min         0.000000       0.000000       0.000000       0.000000   
25%        98.000000     104.000000     109.000000     109.000000   
50%       651.000000     651.000000     653.000000     667.000000   
75%      3326.000000    3758.000000    3874.000000    3929.000000   
max    540292.000000  547386.000000  556606.000000  561816.000000   

             6/15/20  
count     253.000000  
mean    15246.395257  
std     57482.644173  
min         0.000000  
25%       109.000000  
50%       673.000000  
75%      3931.000000  
max    576334.000000  

[8 rows x 148 columns]

In [14]:
recovered_dataset.shape

(253, 150)

In [15]:
deaths_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 266 entries, 0 to 265
Columns: 150 entries, Province/State to 6/15/20
dtypes: float64(2), int64(146), object(2)
memory usage: 311.8+ KB


In [16]:
deaths_dataset.describe()

Lat        Long     1/22/20     1/23/20     1/24/20     1/25/20  \
count  266.000000  266.000000  266.000000  266.000000  266.000000  266.000000   
mean    21.148043   22.538640    0.063910    0.067669    0.097744    0.157895   
std     24.909990   70.466256    1.042337    1.043908    1.473615    2.453621   
min    -51.796300 -135.000000    0.000000    0.000000    0.000000    0.000000   
25%      6.907750  -18.093125    0.000000    0.000000    0.000000    0.000000   
50%     23.488100   20.972650    0.000000    0.000000    0.000000    0.000000   
75%     41.143200   77.191525    0.000000    0.000000    0.000000    0.000000   
max     71.706900  178.065000   17.000000   17.000000   24.000000   40.000000   

          1/26/20     1/27/20     1/28/20     1/29/20  ...         6/6/20  \
count  266.000000  266.000000  266.000000  266.000000  ...     266.000000   
mean     0.210526    0.308271    0.492481    0.500000  ...    1504.161654   
std      3.189730    4.660845    7.664297    7.664793  ...    8215.232193   
min      0.000000    0.000000    0.000000    0.000000  ...       0.000000   
25%      0.000000    0.000000    0.000000    0.000000  ...       2.000000   
50%      0.000000    0.000000    0.000000    0.000000  ...      13.000000   
75%      0.000000    0.000000    0.000000    0.000000  ...     157.000000   
max     52.000000   76.000000  125.000000  125.000000  ...  110124.000000   

              6/7/20         6/8/20         6/9/20        6/10/20  \
count     266.000000     266.000000     266.000000     266.000000   
mean     1514.496241    1528.571429    1546.845865    1566.428571   
std      8250.907923    8292.510696    8373.133522    8452.590717   
min         0.000000       0.000000       0.000000       0.000000   
25%         2.000000       2.000000       2.000000       2.000000   
50%        13.000000      13.500000      13.500000      13.500000   
75%       157.500000     159.000000     159.250000     163.250000   
max    110575.000000  111068.000000  112014.000000  112935.000000   

             6/11/20        6/12/20        6/13/20        6/14/20  \
count     266.000000     266.000000     266.000000     266.000000   
mean     1584.439850    1600.676692    1616.718045    1629.289474   
std      8527.744292    8595.718952    8659.119397    8690.642967   
min         0.000000       0.000000       0.000000       0.000000   
25%         2.000000       2.000000       2.000000       2.000000   
50%        13.500000      14.000000      14.000000      14.000000   
75%       165.500000     166.750000     166.750000     167.000000   
max    113823.000000  114669.000000  115436.000000  115732.000000   

             6/15/20  
count     266.000000  
mean     1642.477444  
std      8728.402797  
min         0.000000  
25%         2.000000  
50%        14.000000  
75%       174.000000  
max    116127.000000  

[8 rows x 148 columns]

In [17]:
deaths_dataset.shape

(266, 150)

**Understanding the data**
- 3 veriseti de 150 sütundan oluşmaktadır.
- Veriler 22 Ocak 2020'den itibaren başlayıp 15 Haziran 2020 tarihine kadar olan verileri içermektedir.
- Country/Region, Lat ve Long sütunları (latitude/longitude, enlem/boylam) verilerin hangi ülkeye ait olduğunu göstermektedir.
- Veri setlerindeki satırları incelediğimizde vaka ve ölüm verisetlerinin satır sayıları aynı iken, iyileşen verisetinde daha az satır olduğunu gözlemliyoruz.

In [18]:
# Tekrar eden satırları kontrol et
print("Tekrar eden satırlar Enfekte olan veri setinde:")
total_duplicates_infected = infected_dataset.duplicated().sum()
print(total_duplicates_infected)

print("\nTekrar eden satırlar İyileşen veri setinde:")
total_duplicates_recovered = recovered_dataset.duplicated().sum()
print(total_duplicates_recovered)

print("\nTekrar eden satırlar Ölen veri setinde:")
total_duplicates_deaths = deaths_dataset.duplicated().sum()
print(total_duplicates_deaths)

Tekrar eden satırlar Enfekte olan veri setinde:
0

Tekrar eden satırlar İyileşen veri setinde:
0

Tekrar eden satırlar Ölen veri setinde:
0


In [19]:
# Belirli bir sütundaki benzersiz değerlerin sayısını kontrol etmek için nunique() kullanılır
print("", infected_dataset['Country/Region'].nunique()) # Enfekte olan ülke sayısı
print("", recovered_dataset['Country/Region'].nunique()) # İyileşen ülke sayısı
print("", deaths_dataset['Country/Region'].nunique()) # Ölen ülke sayısı

 188
 188
 188


**Check for NULL values if any in each column**

In [20]:
# Boş değerleri kontrol et

In [21]:
print("Boş değerler Enfekte olan veri setinde:")
print(infected_dataset.isnull().sum())

Boş değerler Enfekte olan veri setinde:
Province/State    185
Country/Region      0
Lat                 0
Long                0
1/22/20             0
                 ... 
6/11/20             0
6/12/20             0
6/13/20             0
6/14/20             0
6/15/20             0
Length: 150, dtype: int64


In [22]:
print("Boş değerler İyileşen veri setinde:")
print(recovered_dataset.isnull().sum())

Boş değerler İyileşen veri setinde:
Province/State    186
Country/Region      0
Lat                 0
Long                0
1/22/20             0
                 ... 
6/11/20             0
6/12/20             0
6/13/20             0
6/14/20             0
6/15/20             0
Length: 150, dtype: int64


In [23]:
print("Boş değerler Ölen veri setinde:")
print(deaths_dataset.isnull().sum())

Boş değerler Ölen veri setinde:
Province/State    185
Country/Region      0
Lat                 0
Long                0
1/22/20             0
                 ... 
6/11/20             0
6/12/20             0
6/13/20             0
6/14/20             0
6/15/20             0
Length: 150, dtype: int64


**Check number of duplicated values**

In [24]:
# tekrar eden ülke sayısı kontrol et

In [25]:
print("Tekrar eden ülke sayısı Enfekte olan veri setinde:")
print(infected_dataset['Country/Region'].value_counts())

Tekrar eden ülke sayısı Enfekte olan veri setinde:
Country/Region
China             33
Canada            14
United Kingdom    11
France            11
Australia          8
                  ..
Honduras           1
Hungary            1
Iceland            1
India              1
Lesotho            1
Name: count, Length: 188, dtype: int64


In [26]:
print("\nTekrar eden ülke sayısı İyileşen veri setinde:")
print(recovered_dataset['Country/Region'].value_counts())


Tekrar eden ülke sayısı İyileşen veri setinde:
Country/Region
China             33
United Kingdom    11
France            11
Australia          8
Netherlands        5
                  ..
Guinea             1
Guinea-Bissau      1
Guyana             1
Haiti              1
Lesotho            1
Name: count, Length: 188, dtype: int64


In [27]:
print("\nTekrar eden ülke sayısı Ölen veri setinde:")
print(deaths_dataset['Country/Region'].value_counts())


Tekrar eden ülke sayısı Ölen veri setinde:
Country/Region
China             33
Canada            14
United Kingdom    11
France            11
Australia          8
                  ..
Honduras           1
Hungary            1
Iceland            1
India              1
Lesotho            1
Name: count, Length: 188, dtype: int64


**ANALYZE THE DATA SUCH AS `groupby()`**

In [28]:
# Toplam vaka sayılarını ülkeler bazında göster
total_infected_by_country = infected_dataset.groupby("Country/Region").sum().iloc[:, 3:].max(axis=1)
print("\nToplam Vaka Sayıları Ülkeler Bazında:")
print(total_infected_by_country.sort_values(ascending=False))


Toplam Vaka Sayıları Ülkeler Bazında:
Country/Region
US                  2114026
Brazil               888271
Russia               536484
India                343091
United Kingdom       298315
                     ...   
Seychelles               11
MS Zaandam                9
Western Sahara            9
Papua New Guinea          8
Lesotho                   4
Length: 188, dtype: int64


### 

In [29]:
# Toplam iyileşen sayılarını ülkeler bazında göster
total_recovered_by_country = recovered_dataset.groupby("Country/Region").sum().iloc[:, 3:].max(axis=1)
print("\nToplam İyileşen Sayıları Ülkeler Bazında:")
print(total_recovered_by_country.sort_values(ascending=False))


Toplam İyileşen Sayıları Ülkeler Bazında:
Country/Region
US                  576334
Brazil              477709
Russia              284021
India               180013
Italy               177010
                     ...  
Papua New Guinea         8
Western Sahara           8
Lesotho                  2
Sweden                   0
MS Zaandam               0
Length: 188, dtype: int64


In [30]:
# Toplam ölüm sayılarını ülkeler bazında göster
total_deaths_by_country = deaths_dataset.groupby("Country/Region").sum().iloc[:, 3:].max(axis=1)
print("\nToplam Ölüm Sayıları Ülkeler Bazında:")
print(total_deaths_by_country.sort_values(ascending=False))


Toplam Ölüm Sayıları Ülkeler Bazında:
Country/Region
US                                  116127
Brazil                               43959
United Kingdom                       41821
Italy                                34371
France                               29439
                                     ...  
Saint Lucia                              0
Cambodia                                 0
Saint Vincent and the Grenadines         0
Eritrea                                  0
Dominica                                 0
Length: 188, dtype: int64


**Total cases, recoveries and deaths by country**

In [31]:
# Ülkeler bazında toplam vaka, iyileşen ve ölüm sayılarını göster
total_infected_recovered_deaths_by_country = pd.concat([total_infected_by_country, total_recovered_by_country, total_deaths_by_country], axis=1)
total_infected_recovered_deaths_by_country.columns = ["Toplam Vaka", "Toplam İyileşen", "Toplam Ölüm"]
print("\nÜlkeler Bazında Toplam Vaka, İyileşen ve Ölüm Sayıları:")
print(total_infected_recovered_deaths_by_country.sort_values(by="Toplam Vaka", ascending=False))


Ülkeler Bazında Toplam Vaka, İyileşen ve Ölüm Sayıları:
                  Toplam Vaka  Toplam İyileşen  Toplam Ölüm
Country/Region                                             
US                    2114026           576334       116127
Brazil                 888271           477709        43959
Russia                 536484           284021         7081
India                  343091           180013         9900
United Kingdom         298315             1284        41821
...                       ...              ...          ...
Seychelles                 11               11            0
MS Zaandam                  9                0            2
Western Sahara              9                8            1
Papua New Guinea            8                8            0
Lesotho                     4                2            0

[188 rows x 3 columns]
